users/users.go

```diff
@@ -54,16 +61,22 @@ func (u User) Validate() error {
 type UserRepository interface {
 	// Save persists the user account. A non-nil error is returned to indicate
 	// operation failure.
-	Save(ctx context.Context, u User) error
+	Save(ctx context.Context, u User) (User, error)
```

Comment:
Why do we need to return the User? We usually only need an ID of the created entity.